### DISCUSSION TITLE
EOL PHP 7.4

### DISCUSSION THREAD

**Discussion by K8Sudo** [monarc-project/MonarcAppFO#469] [Q&A] [answered]
Is there a plan to release an update before the 28th of November, which brings support for php 8.0 or 8.1? Currently composer still expects ^7.4. Or am I missing something?

[https://www.php.net/supported-versions.php](https://www.php.net/supported-versions.php)

**Comment by ruslanbaidan** [comment]
For the moment it's not possible due to some parts of the code.
We are working on refactoring that should allow this.
But this is for the next year.

**Reply by K8Sudo** [reply]
So, as of November 28, we will have to work with a potentially vulnerable PHP version that is no longer provided with security updates? Is there already an estimate for next year?

**Reply by ruslanbaidan** [reply]
What is the exact PHP version you use on your Monarc server(s) ?

**Reply by K8Sudo** [reply]
We use the latest version 7.4.33 from November 3rd.

**Reply by ruslanbaidan** [reply]
There is no reason to worry about data leaking or that someone could hack the application due to the security problems of php version. If you followed the changelog of php 7.4.x there were no core issues that could lead to that and we use only reliable dependencies packages.
The plan is of course, to migrate Monarc to php8.x as soon as possible, this is also due to the performance improvements.

**Comment by jerolomb** [comment]
There is now a branch (feature/php8) with php8 you can switch your FO on it, tested for myself with php 8.0.25

**Reply by K8Sudo** [reply]
In my test installation I get an error when running the update-all script. The error occurs when running `grunt concat`. Here the Angular modules are not found.

```
Running "concat:angularCommonLibsJs" (concat) task
>> Source file "node_modules/jquery/dist/jquery.min.js" not found.
>> Source file "node_modules/angular/angular.min.js" not found.
>> Source file "node_modules/angular-animate/angular-animate.min.js" not found.
>> Source file "node_modules/angular-aria/angular-aria.min.js" not found.
>> Source file "node_modules/angular-gettext/dist/angular-gettext.min.js" not found.
>> Source file "node_modules/angular-material/angular-material.min.js" not found.
>> Source file "node_modules/@uirouter/angularjs/release/angular-ui-router.min.js" not found.
>> Source file "node_modules/angular-resource/angular-resource.min.js" not found.
>> Source file "node_modules/angular-local-storage/dist/angular-local-storage.min.js" not found.
>> Source file "node_modules/angular-material-data-table/dist/md-data-table.min.js" not found.
>> Source file "node_modules/angular-breadcrumb/release/angular-breadcrumb.min.js" not found.
...
```

**Reply by ruslanbaidan** [reply]
The error that you wrote is not related to the specific of the branch.
The code adjustments were done only for the Backend and the errors are with Frontend modules.
You can try to install a separate FrontOffice instance to test.
It's recommended execute all the possible tests before using the version on production and would be great if you could help us with the tests.

**Reply by K8Sudo** [reply]
The installation I used here is a completely new virtual machine running Ubuntu 20.04 with PHP 8.0.25 and NodeJS 16.

The installation completely follows the installation instructions except for the git clone. `git clone --branch feature/php8 https://github.com/monarc-project/MonarcAppFO.git`

**Reply by K8Sudo** [reply]
I will give it another try next week and let you know again

**Reply by K8Sudo** [reply]
We have now tested again extensively. We can confirm that the feature branch works under PHP 8.0 as well as 8.1. We have tested on an Ubuntu 18.04, 20.04 and 22.04. The above mentioned problems with the frontend were caused by the fact that we also wanted to update the nodejs version to the latest LTS version 18.13.0. However, there are still problems here. 16.x and 17.x work fine.

**Reply by ruslanbaidan** [reply]
Thank you for the feedback.
I think we can plan php8 support release for the next time.
We are going to test the backward compatibility (if the app works well with php7.4 on the php8 branch).

**Reply by ruslanbaidan** [reply, marked as answer]
From the version of 2.12.6 the FrontOffice and BackOffice apps work on php8.
